contactsummary/js/src/contact-summary/services/contract.js

```diff
@@ -75,15 +75,9 @@ define([
          */
         factory.getPrimary = function () {
             return this.get().then(function (response) {
-                var primary = {};
+                var sortedContracts = _.sortBy(response, ['end_date', 'is_primary'], ['desc', 'desc']);
 
-                angular.forEach(response, function (contract) {
-                    if (contract.is_primary === '1' && contract.is_current === '1') {
```

Comment:
Ok, please add document what constitutes a primary contract in the function description